components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -4845,6 +4846,7 @@ public void deleteApplicationByResourceId(String resourceId,
 
         try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
             ServiceProvider application = getApplicationByResourceId(resourceId, tenantDomain);
+            int appIdUsingResourceId = getAppIdUsingResourceId(resourceId, tenantDomain);
```

Comment:
can't we use `application.getApplicationID()` instead?

Reply:
Fixed with 43ba30a50e3dda59495ca8caba174e8259b9122c